app/models/course/level.rb

```diff
@@ -1,9 +1,26 @@
 class Course::Level < ActiveRecord::Base
-  validates :experience_points_threshold, numericality: { greater_than: 0 }
+  validates :experience_points_threshold, numericality: { greater_than_or_equal_to: 0 }
+  validates :experience_points_threshold, uniqueness: { scope: :course_id }
```

Comment:
yes, validations are generated from the database constraints by schema_validations.